services/api/src/resources/insight/resolvers.ts

```diff
@@ -0,0 +1,169 @@
+  endpoint: config.origin,
+  accessKeyId: config.accessKeyId,
+  secretAccessKey: config.secretAccessKey,
+  region: config.region,
+  params: {
+    Bucket: config.bucket
+  },
+  s3ForcePathStyle: true,
+  signatureVersion: 'v4'
+});
+
+const convertDateFormat = R.init;
+
+
+// Get insights files directly from the bucket
+export const getInsightsBucketFiles = async ({ prefix }) => {
+	try {
+    const data = await s3Client.listObjects({ Bucket: bucket, Prefix: prefix }, function(err, data) {
+      if (err) {
+        console.log("Failed to get items: ", err);
+      } else {
+
+         return data;
+      }
+    }).promise();
+
+    if (!data) {
+      return null;
+    }
+
+    return await JSON.parse(JSON.stringify(data.Contents));
+	}
+  catch (e) {
+    throw new Error(`Error retrieving bucket items - ${e.Error}`)
+	}
+}
+
+export const getInsightsDownloadUrl: ResolverFn = async (
+  { fileId, environment, file },
+  _args,
+  { sqlClientPool }
+) => {
+
+  const environmentData = await environmentHelpers(
+    sqlClientPool
+  ).getEnvironmentById(parseInt(environment));
+  const projectData = await projectHelpers(sqlClientPool).getProjectById(
+    environmentData.project
+  );
+
+  let environmentName = getEnvironmentName(environmentData, projectData);
+
+	try {
+    const s3Key = `insights/${projectData.name}/${environmentName}/${file}`;
+
+    return s3Client.getSignedUrl('getObject', {Bucket: bucket, Key: s3Key, Expires: 600});
+	} catch (e) {
+   return `Error while creating download link - ${e.Error}`
+	}
+}
+
+export const getInsightsFileData: ResolverFn = async (
+  { fileId, environment, file },
+  _args,
+  { sqlClientPool }
+) => {
+  if (!fileId) {
+    return null;
+  }
+
+  const environmentData = await environmentHelpers(
+    sqlClientPool
+  ).getEnvironmentById(parseInt(environment));
+  const projectData = await projectHelpers(sqlClientPool).getProjectById(
+    environmentData.project
+  );
+
+  let environmentName = getEnvironmentName(environmentData, projectData);
+
+  try {
+    let insightsFile = 'insights/'+projectData.name+'/'+environmentName+'/'+file
+    const data = await s3Client.getObject({Bucket: bucket, Key: insightsFile}).promise();
+
+    if (!data) {
+      return null;
+    }
+
+    if (data.ContentEncoding == 'gzip') {
+      return "Compressed file"
+    }
+    return JSON.parse(JSON.stringify(data.Body));
+  } catch (e) {
+    return `There was an error loading insights data: ${e.message}\nIf this error persists, contact your Lagoon support team.`;
+  }
+};
+
+export const getInsightsFilesByEnvironmentId: ResolverFn = async (
+  { id: eid, environmentAuthz },
+  { name, limit },
+  { sqlClientPool, hasPermission }
+) => {
+
+  if (!eid) {
+    throw "No Environment ID given.";
+  }
+
+  const environmentData = await environmentHelpers(
+    sqlClientPool
+  ).getEnvironmentById(parseInt(eid));
+
+  if (!environmentAuthz) {
+    await hasPermission('environment', 'view', {
```

Comment:
I wonder if this shouldn't require elevated perms?

Reply:
this check only happens if a permissions check hasn't already happened in the environment resolver